src/components/RegisterForm/index.tsx

```diff
@@ -0,0 +1,224 @@
+						/>
+					</FormItem>
+					<Errors>
+						{errors.email && <p>{errors.email.message}</p>}
+					</Errors>
+
+					{/* NICKNAME */}
+					<FormItem>
+						<Label htmlFor="nickname">Přezdívka:</Label>
+						<FormInput
+							name="nickname"
+							type="text"
+							placeholder="Přezdívka1"
+							ref={register({
+								required: 'Musíte zadat přezdívku!',
+								maxLength: {
+									value: 30,
+									message:
+										'Vaše přezdívka má více než 30 znaků!',
+								},
+								// TODO: Must be unique!
+							})}
+						/>
+					</FormItem>
+					<Errors>
+						{errors.nickname && <p>{errors.nickname.message}</p>}
+					</Errors>
+
+					{/* PASSWORD */}
+					<FormItem>
+						<Label htmlFor="password">Heslo:</Label>
+						<FormInput
+							name="password"
+							type="password"
+							placeholder="1SafePassword%"
+							onFocus={() => setHidden(false)} // enables PasswordMessage
+							ref={register({
+								// messages not necessary (PasswordMessage handles those)
+								required: '',
+								pattern: {
+									value: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{6,}$/,
+									message: '',
+								},
+								minLength: {
+									value: 6,
+									message: '',
+								},
+							})}
+						/>
+					</FormItem>
+					<HiddenError>
+						{errors.password && <p>{errors.password.message}</p>}
+					</HiddenError>
+
+					<PasswordMessage hidden={hidden}>
+						<ValidPassword validPassword={hasLowerCase}>
+							malé písmeno
+						</ValidPassword>
+						<ValidPassword validPassword={hasMinimumLength}>
+							více než 6 znaků
+						</ValidPassword>
+						<ValidPassword validPassword={hasUpperCase}>
+							VELKÉ PÍSMENO
+						</ValidPassword>
+						<ValidPassword validPassword={hasNumber}>
+							číslo
+						</ValidPassword>
+					</PasswordMessage>
+
+					{/* CONFIRM PASSWORD */}
+					<FormItem>
+						<Label htmlFor="passwordConfirm">Potvrď heslo:</Label>
+						<FormInput
+							name="passwordConfirm"
+							type="password"
+							placeholder="1SafePassword%"
+							ref={register({
+								validate: (value) =>
+									value === password.current ||
+									'Hesla se neshodují!',
+							})}
+						/>
+					</FormItem>
+					<Errors>
+						{errors.passwordConfirm && (
+							<p>{errors.passwordConfirm.message}</p>
+						)}
+					</Errors>
+
+					{/* CHECK */}
+					<FormItem>
+						<Label htmlFor="check">Souhlasím s XY:</Label>
+						<FormCheckbox
+							name="check"
+							type="checkbox"
+							ref={register({
+								required:
+									'Registrace bez souhlasu s podmínkami není možná!',
+							})}
+						/>
+					</FormItem>
+					<Errors>
+						{errors.check && <p>{errors.check.message}</p>}
+					</Errors>
+
+					{/* SUBMIT */}
+					<FormItem>
+						<FormButton name="submit" type="submit">
+							ODESLAT
+						</FormButton>
+					</FormItem>
+				</Form>
+			</Route>
+
+			<Route path="/register/successful">
```

Comment:
Do we want to reflect this successful state in URL address? :thinking:

Reply:
I think it doesn't hurt anyone and it makes the success very clear. :joy: